core-common/src/main/java/org/mapstruct/factory/Mappers.java

```diff
@@ -93,11 +97,13 @@ private Mappers() {
         throw new ClassNotFoundException("Cannot find implementation for " + mapperType.getName() );
     }
 
-    private static <T> T doGetMapper(Class<T> clazz, ClassLoader classLoader) {
+    private static <T> T doGetMapper(Class<T> clazz, ClassLoader classLoader) throws NoSuchMethodException {
         try {
             @SuppressWarnings("unchecked")
-            T mapper = (T) classLoader.loadClass( clazz.getName() + IMPLEMENTATION_SUFFIX ).newInstance();
-            return mapper;
+            Class<T> implementation = (Class<T>) classLoader.loadClass( clazz.getName() + IMPLEMENTATION_SUFFIX );
+            Constructor<T> constructor = implementation.getDeclaredConstructor((Class<?>[]) null);
+            constructor.setAccessible(true);
```

Comment:
I think on Java 9 and beyon we should use `trySetAccessible()` instead. This will allow to raise a meaningful exception in case we can't make it accessible due to lack of `opens`.

Reply:
I did not have a good knowledge of Java9 but I agree.  Do you already move your code to Java9 ?

Reply:
We should _support_ Java 9 as well as possible, but not _require_ it. I.e. I'd suggest to do it like this:

```
if ( isJava9 ) {
     trySetAccessible;
}
else {
    setAccessible;
}